model/src/main/proto/interaction_object.proto

```diff
@@ -0,0 +1,69 @@
+syntax = "proto3";
+
+package model;
+
+option java_package = "org.oppia.app.model";
+option java_multiple_files = true;
+
+message InteractionObject {
+  oneof object {
+    int32 int_value = 1;
+    string string_value = 2;
+    bool bool_value = 3;
+    // repeated fields cannot be within a oneof so lists are
+    // wrapped into this parameter.
+    List list = 4;
+    Graph graph = 5;
+    ImageWithRegions image_with_regions = 6;
+    LogicQuestion logic_question = 7;
+  }
+}
+
+// Structure containing lists. Note we should fill only
+// one of the fields in this message.
+// We do not add a oneof here because repeated fields cannot be within a oneof.
+message List {
+  repeated string string_list = 1;
+}
+
+// Structure representing a graph object.
+message Graph {
```

Comment:
See above--let's only focus on the prototype interactions for now: #28->#35.

Reply:
Done